compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -381,7 +381,7 @@ where
     result
 }
 
-pub(crate) fn needs_normalization<'tcx, T: TypeVisitable<'tcx>>(value: &T, reveal: Reveal) -> bool {
+pub fn needs_normalization<'tcx, T: TypeVisitable<'tcx>>(value: &T, reveal: Reveal) -> bool {
```

Comment:
that shouldn't be exported, this is only used by the visitors to check whether they should recurse. Use `<T as TypeVisitable>::has_projections` instead.